swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -529,26 +675,13 @@ def initial_setup(
             any axis, in units of particle distance along that axis. Is only used
             if ``method = 'displaced'``
 
-
-
         Notes
         -----
 
-            + Must be called AFTER the ``self.run_params`` paramters have been 
-              tweaked by the user.
+        + Must be called AFTER the ``self.run_params`` paramters have been 
+        tweaked by the user.
 
         """
-        # do this again in case some brainiac has the
-        # bright idea to change it in the meantime
-        self.npart = self.number_of_particles ** self.ndim
-        self.boxsize_to_use = self.boxsize.to(self.unit_system["length"]).value
```

Comment:
I'm worried about this specifically because of the comment above.

Reply:
Because this is now a @property it's calculated every time it's used

Reply:
allrighty then